cocos/audio/android/AudioDecoder.cpp

```diff
@@ -468,8 +477,11 @@ bool AudioDecoder::decodeToPcm()
 
     ALOGV("Stopped decoding");
 
-    /* Destroy the UrlAudioPlayer object */
-    SL_DESTROY_OBJ(_playObj);
+    /* Destroy the AudioPlayer object */
+    {
+        std::lock_guard<std::mutex> lk(__SLPlayerMutex);
+        SL_DESTROY_OBJ(_playObj);
+    }
```

Comment:
there are 3 diff's. Two doesn’t work, and 1 work.
[doesnt-work1-pull17049.diff.txt](https://github.com/cocos2d/cocos2d-x/files/666633/doesnt-work1-pull17049.diff.txt)
[doesnt-work2-pull17049.diff.txt](https://github.com/cocos2d/cocos2d-x/files/666637/doesnt-work2-pull17049.diff.txt)
[work-pull17049.diff.txt](https://github.com/cocos2d/cocos2d-x/files/666639/work-pull17049.diff.txt)